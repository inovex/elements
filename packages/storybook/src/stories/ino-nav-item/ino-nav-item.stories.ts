import { html } from 'lit-html';

export default {
  title: `Structure/ino-nav-item`,
  component: 'ino-nav-item',
};

export const Playground = args => html`
  <div class="story-list">
    <h4>Customizable nav item</h4>
    <ino-list>
      <ino-nav-item
        class="customizable-nav-item"
        text="${args.text}"
        sub-text="${args.subText}"
        activated="${args.activated}"
        disabled="${args.disabled}"
      >
        <ino-icon icon="onboarding"></ino-icon>
      </ino-nav-item>
    </ino-list>
`;
Playground.args = {
  text: 'Nav item',
  subText: 'Nav item sub text',
  activated: false,
  disabled: false
};

export const States = () => html`
  <ino-list>
    <ino-nav-item text="Simple item"></ino-nav-item>
    <ino-nav-item text="Simple item 2"></ino-nav-item>
    <ino-nav-item activated text="Activated item"></ino-nav-item>
    <ino-nav-item disabled text="Disabled item"></ino-nav-item>
  </ino-list>
`;

export const TwoLines = () => html`
  <ino-list two-lines="true">
    <ino-nav-item text="Simple item" sub-text="Secondary Text"></ino-nav-item>
    <ino-nav-item text="Simple item 2" sub-text="Secondary Text 2"></ino-nav-item>
  </ino-list>
`;

export const Graphic = () => html`
  <ino-list>
    <ino-nav-item text="Lorem ipsum dolor sit">
      <ino-icon icon="onboarding"></ino-icon>
    </ino-nav-item>
    <ino-nav-item text="Lorem ipsum dolor sit">
      <ino-img src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png" ratio-width="1" ratio-height="1"></ino-img>
    </ino-nav-item>
  </ino-list>
  </div>
`;
