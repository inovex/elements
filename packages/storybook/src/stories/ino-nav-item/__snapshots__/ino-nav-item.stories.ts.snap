// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-nav-item Activated smoke-test 1`] = `
<div id="root-inner">
  <div class="story-list">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-nav-item class="customizable-nav-item hydrated"
                      text="Nav item"
                      sub-text
                      activated="true"
                      disabled="false"
        >
          <ino-list-item class="hydrated">
            <span class
                  slot
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-deprecated-list-item--activated mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__text">
                Nav item
              </span>
            </li>
          </ino-list-item>
        </ino-nav-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-nav-item Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-list">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-nav-item class="customizable-nav-item hydrated"
                      text="Nav item"
                      sub-text
                      activated="false"
                      disabled="true"
        >
          <ino-list-item class="hydrated">
            <span class
                  slot
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-deprecated-list-item--disabled mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__text">
                Nav item
              </span>
            </li>
          </ino-list-item>
        </ino-nav-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-nav-item Graphic smoke-test 1`] = `
<div id="root-inner">
  <div class="story-list">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-nav-item class="customizable-nav-item hydrated"
                      text="Nav item"
                      sub-text
                      activated="false"
                      disabled="false"
        >
          <ino-list-item class="hydrated">
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <span class="ino-nav-item--leading-slot"
                      slot="leading"
                >
                  <ino-icon icon="onboarding"
                            class="ino-icon--color-scheme-primary hydrated"
                  >
                  </ino-icon>
                </span>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Nav item
              </span>
            </li>
          </ino-list-item>
        </ino-nav-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-nav-item Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-list">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-nav-item class="customizable-nav-item hydrated"
                      text="Nav item"
                      sub-text
                      activated="false"
                      disabled="false"
        >
          <ino-list-item class="hydrated">
            <li class="mdc-deprecated-list-item"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <span class="ino-nav-item--leading-slot"
                      slot="leading"
                >
                  <ino-icon icon="onboarding"
                            class="hydrated"
                  >
                  </ino-icon>
                </span>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Nav item
              </span>
            </li>
          </ino-list-item>
        </ino-nav-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-nav-item Text smoke-test 1`] = `
<div id="root-inner">
  <div class="story-list">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-nav-item class="customizable-nav-item hydrated"
                      text="Some Text"
                      sub-text
                      activated="false"
                      disabled="false"
        >
          <ino-list-item class="hydrated">
            <span class
                  slot
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__text">
                Some Text
              </span>
            </li>
          </ino-list-item>
        </ino-nav-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-nav-item TwoLines smoke-test 1`] = `
<div id="root-inner">
  <div class="story-list">
    <ino-list class="hydrated">
      <ul class="mdc-deprecated-list">
        <ino-nav-item class="customizable-nav-item hydrated"
                      text="Nav item"
                      sub-text="Secondary Text 2"
                      activated="false"
                      disabled="false"
        >
          <ino-list-item class="hydrated">
            <span class
                  slot
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__text">
                <span class="mdc-deprecated-list-item__primary-text">
                  Nav item
                </span>
                <span class="mdc-deprecated-list-item__secondary-text">
                  Secondary Text 2
                </span>
              </span>
            </li>
          </ino-list-item>
        </ino-nav-item>
      </ul>
    </ino-list>
  </div>
</div>
`;
